ddtrace/llmobs/_evaluations/ragas/faithfulness/evaluator.py

```diff
@@ -0,0 +1,85 @@
+import atexit
+from concurrent import futures
+import math
+import time
+from typing import Dict
+from typing import Optional
+
+from ddtrace import config
+from ddtrace.internal import forksafe
+from ddtrace.internal.logger import get_logger
+from ddtrace.internal.periodic import PeriodicService
+
+from ...._writer import LLMObsEvaluationMetricEvent
+
+
+logger = get_logger(__name__)
+
+
+class RagasFaithfulnessEvaluator(PeriodicService):
+    """Base class for evaluating LLM Observability span events"""
+
+    def __init__(self, interval: float, _evaluation_metric_writer=None, _llmobs_instance=None):
+        super(RagasFaithfulnessEvaluator, self).__init__(interval=interval)
+        self.llmobs_instance = _llmobs_instance
+        self._lock = forksafe.RLock()
```

Comment:
I'm slightly concerned that this will lead to tons of new background threads being spawned/active each with their own locks, which increases the likelihood of introducing some race condition.

Is it possible to have a single evaluator service handle all of the evaluations? Is having a separate thread per individual evaluation metric viable? I'd like to chat some about this before we merge this

Reply:
This is a good point, will schedule a time to talk about this